services/blockstorage/sync/state_waiting_for_chunks_test.go

```diff
@@ -40,16 +40,17 @@ func TestWaitingMovesToIdleOnTimeout(t *testing.T) {
 }
 
 func TestWaitingAcceptsNewBlockAndMovesToProcessing(t *testing.T) {
+	manualWaitForChunksTimer := synchronization.NewTimerWithManualTick()
 	blocksMessage := builders.BlockSyncResponseInput().Build().Message
-	h := newBlockSyncHarness().withNodeKey(blocksMessage.Sender.SenderPublicKey()).withWaitForChunksTimeout(time.Second)
+	h := newBlockSyncHarnessWithManualWaitForChunksTimeoutTimer(manualWaitForChunksTimer).withNodeKey(blocksMessage.Sender.SenderPublicKey())
 
-	h.storage.When("LastCommittedBlockHeight").Return(primitives.BlockHeight(10)).Times(1)
+	h.expectLastCommittedBlockHeight(primitives.BlockHeight(10))
 	h.gossip.When("SendBlockSyncRequest", mock.Any, mock.Any).Return(nil, nil).Times(1)
 
-	waitingState := h.sf.CreateWaitingForChunksState(h.config.NodePublicKey())
-	nextState := h.nextState(waitingState, func() {
-		time.Sleep(10 * time.Millisecond) // yield to make sure the state is fully 'running', cannot latch the internal goroutine
+	waitingState := h.factory.CreateWaitingForChunksState(h.config.NodePublicKey())
+	nextState := h.processStateAndWaitUntilFinished(waitingState, func() {
 		waitingState.gotBlocks(h.ctx, blocksMessage)
+		manualWaitForChunksTimer.ManualTick()
```

Comment:
this is not required here, the sleep above should go because the issue it was solved was resolved using the conduit, but not sure there is any need for the manual latch here, lets go over this together

Reply:
I'll add // for completion